### ISSUE TITLE
Better documentation of flutter install flags and side-effects

### ISSUE THREAD

**Issue by mazei513** [flutter/flutter#40560]
## Steps to Reproduce

1. [main.dart](https://gist.github.com/mazei513/cd5dce338323031d2a5e07a800a592f5)
2. `flutter run --release` with a device connected, it should print `true`. Quit out of the `flutter run --release` with `q`.
3. `flutter run --release` again, and it will print `false`. Quit out of the `flutter run --release` with `q`.
4. `flutter build apk && flutter install`, then run the app on the device. It prints `true`.
5. `flutter build apk && flutter install` again, then run the app on the device. It prints `true` again.

## Expectations

From my understanding, SQFLite DB would not be lost between installations, which results in the expected output when using `flutter run --release`. However, when updating the app through `flutter install` the DB file is lost, presumingly during the `Uninstalling old version...` step. Is there a way to perform `flutter install` without the uninstall step?

## Context

Developing an app to be used personally only on my phone. Would like to keep the install size small by splitting the apk with `flutter build`.

## Doctor

```
[✓] Flutter (Channel stable, v1.9.1+hotfix.2, on Mac OS X 10.14.6 18G95, locale en-GB)
    • Flutter version 1.9.1+hotfix.2 at /Users/[user_here]/flutter/flutter
    • Framework revision 2d2a1ffec9 (9 days ago), 2019-09-06 18:39:49 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/[user_here]/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.3, Build version 10G8
    • CocoaPods version 1.7.5

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 38.2.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.38.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.4.1

[✓] Connected device (1 available)
    • Mi A1 • e4916a3d0504 • android-arm64 • Android 9 (API 28)
```

**Comment by mannprerak2** [comment]
Why not use ```flutter run --release``` ?
It will also create an output APK file just like ```flutter build apk```

If you'd like to install an APK without building it again. I suggest you use
``` adb install /path/to/your/apk/file.apk ```

**Comment by mazei513** [comment]
> Why not use `flutter run --release` ?
> It will also create an output APK file just like `flutter build apk`

Well specifically I wanted to do `flutter build apk` with `--split-per-abi` to have the built apk be smaller.

> If you'd like to install an APK without building it again. I suggest you use
> `adb install /path/to/your/apk/file.apk`

This sounds like it will achieve what I want, will try it out, thank you.

**Comment by escamoteur** [comment]
I'm not sure if he have covered enough in the documentation when an app is installed with deleting all data and when not.

**Comment by mazei513** [comment]
`adb install /path/to/your/apk/file.apk` managed to do it. Though perhaps could put it up somewhere in the docs that running `flutter install` may cause your app's previous data to be deleted. Or perhaps add a `--skip-uninstall` flag to `flutter install` perhaps?

**Comment by adminant** [comment]
`--skip-uninstall` flag wold be very helpful for `flutter install` command. Because now one forced to enter all credentials as for 1st run after every `flutter install`.

**Comment by adminant** [comment]
`--skip-uninstall` flag wold be very helpful for `flutter install` command. Because now one forced to enter all credentials as for 1st run after every `flutter install`.

**Comment by bksubhuti** [comment]
Yes.. i'm also looking for a --skip-uninstall because we need to make sure the database gets copied over.  When testing on my phone, it always uninstalls.. therefore it always copies the database over.
